src/adapter-types/src/dyncfgs.rs

```diff
@@ -18,7 +18,17 @@ pub const ENABLE_STATEMENT_LIFECYCLE_LOGGING: Config<bool> = Config::new(
     "Enable logging of statement lifecycle events in mz_internal.mz_statement_lifecycle_history.",
 );
 
+/// Whether the default sink partitioning strategy for an environment should be 'v1'. When set to
+/// false the strategy defaults to 'v0'.
+pub const ENABLE_DEFAULT_V1_PARTITION_STRATEGY: Config<bool> = Config::new(
```

Comment:
Should this turn into a string that accepts values {v0, v1}?

Reply:
This will force me to handle the case where the string provided in the config is invalid. What should the planner do in this case? Reject the query or choose some other default?

Reply:
Indeed it would! Not sure: what do you think is best? Rejecting the query seems reasonable to me!

Reply:
necrobumping, but done!